core/services/ocr2/plugins/median/plugin.go

```diff
@@ -111,33 +121,24 @@ func NewMedianServices(ctx context.Context,
 			abort()
 			return
 		}
-		median = loop.NewMedianService(lggr, func() *exec.Cmd {
+		median := loop.NewMedianService(lggr, func() *exec.Cmd {
 			cmd := exec.Command(cmdName)
 			plugins.SetCmdEnvFromConfig(cmd, registeredLoop.EnvCfg)
 			return cmd
-		})
+		}, provider, dataSource, juelsPerFeeCoinSource, errorLog)
+		argsNoPlugin.ReportingPluginFactory = median
+		srvs = append(srvs, median)
 	} else {
-		median = NewPlugin(lggr)
-	}
-	srvs = append(srvs, median)
-	argsNoPlugin.ReportingPluginFactory, err = median.NewMedianFactory(ctx, provider, ocrcommon.NewDataSourceV2(pipelineRunner,
```

Comment:
We were getting blocked here, because the `MedianService` is not live until started, and not started until after returning. Now the MedianService is a ReportingPluginFactory, and this internal call is included up front when refreshing the client, just like we do with the `RelayerService`.